### ISSUE TITLE
Milvus: support hybrid search

### ISSUE THREAD

**Issue by langchain4j** [langchain4j/langchain4j#2299]
https://milvus.io/blog/introduce-milvus-2-5-full-text-search-powerful-metadata-filtering-and-more.md

**Comment by asllani94** [comment]
I really want to see this feature added to langchain4j. None of major Java LLM frameworks support this as of now but it is available in the Milvus Java client

**Comment by yangzho12138** [comment]
@dliubarskyi Hi, do we still consider implementing this feature? If so, I would like to take a look, thanks!

**Comment by dliubarskyi** [comment]
@yangzho12138 yes, please go ahead!

**Comment by yangzho12138** [comment]
@dliubarskyi Right now, only the V2 service API supports the built-in BM25 function for sparse vectors. Shall we update all the related services to V2, or just implement a BM25 algorithm on our side for now? Thanks for the suggestion!

**Comment by dliubarskyi** [comment]
@yangzho12138 not sure I understand, we are using Milvus SDK v2.5.7, so it should be supported, no?

**Comment by yangzho12138** [comment]
@dliubarskyi Yes, we can support V2 service API based on the version, but right now we do not use it in our implementation (We use MilvusServiceClient, now it could use MilvusClientV2)
So I'm just wondering whether need to update all to V2 service API, or just keep v1 implementation.

**Comment by dliubarskyi** [comment]
@yangzho12138 thanks for explanation! Do you know if `MilvusServiceClient` is deprecated and only `MilvusClientV2` will be supported in the future? If yes, then I would migrate completely to `MilvusClientV2`

**Comment by yangzho12138** [comment]
@dliubarskyi https://docs.zilliz.com/reference/java/v1-Connections-MilvusClient I think so, please refer to this link, marked as "about to deprecated". I can try to migrate to V2 service.

**Comment by dliubarskyi** [comment]
@yangzho12138 then it makes sense to migrate to V2 for sure, thank you for the pointer!

**Comment by asllani94** [comment]
@yangzho12138 In order to support hybrid search a change in EmbeddingSearchRequest model needs to be made. Right now the implementation of this class only support field queryEmbedding in as a float vector, in order to perform BM25 search and vector search together, EmbeddingSearchRequest needs to also pass to the EmbeddingStore also the query in plain text. I have a private project where I have extended EmbeddingSearchRequest and MilvusEmbeddingStore as well to make it work with V2 client. EmbeddingSearchRequest is located in langchain4j-core.

**Comment by 110LK** [comment]
I have experience implementing mixed search using langchain4j milvus in my personal projects. I will try this issue.😊

**Comment by yangzho12138** [comment]
@110LK I almost finish this, writing tests for it

**Comment by 110LK** [comment]
@yangzho12138 I looked at your code, it's much better than mine!👍It's a pity I lost a chance to contribute.😂

**Comment by yangzho12138** [comment]
@110LK I'm pretty sure u can find another interesting topic to contribute! (your message notified me to finish this hhh, too busy recently, almost forgot this 😂) Looking forward to your contribution!
